### PR TITLE
Refined Readme and added chart for depended Casablanca version

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,7 +78,21 @@ Starting from version 2.1.0, Azure Storage Client Library for C++ supports Visua
 
 ### C++ REST SDK
 
-The Azure Storage Client Library for C++ depends on the C++ REST SDK (codename "Casablanca") 2.10.3. It can be installed through Vcpkg (vcpkg install cpprestsdk) or downloaded directly from [GitHub](https://github.com/Microsoft/cpprestsdk/releases/tag/v2.10.3).
+The Azure Storage Client Library for C++ depends on the C++ REST SDK (codename "Casablanca") It can be installed through Vcpkg (vcpkg install cpprestsdk) or downloaded directly from [GitHub](https://github.com/Microsoft/cpprestsdk/releases/).
+
+The dependency version can be found in the following chart:
```
Comment (vinjiang) [inline README.md]: This matrix is to give customer the configuration we confirmed can work. How about 'validated matrix'?

File: README.md [hunk 2]
```diff
@@ -151,7 +165,7 @@ vi ../../samples/SamplesCommon/samples_common.h # modify connection string to in
 
 Please note the current build script is only tested on Ubuntu 16.04. Please update the script accordingly for other distributions.
 
-Please note that starting from 2.10.0, Casablanca requires a minimum version of CMake v3.1, so the default CMake on Ubuntu 14.04 cannot support Casablanca build. User can upgrade CMake by themselves to build Casablanca.
+Please note that starting from 2.10.0, Casablanca requires a minimum version of CMake v3.1, so the default CMake on Ubuntu 14.04 cannot support Casablanca build. User can upgrade CMake by themselves to build Casablanca. 5.0.1 is the last version that uses Casablanca 2.9.1, which supports default Ubuntu 14.04's CMake.
```
Comment (vinjiang) [inline README.md]: last version or latest version?

File: README.md (hunk 1 above)
Comment (katmsft) [reply README.md]: True, changed to `validated Casablanca version for each major or recent release on different platforms`

File: README.md (hunk 2 above)
Comment (katmsft) [reply README.md]: Actually, this is mis-leading. This line is refined.

**Pull request merged**